dbms/src/Operators/LimitTransformOp.cpp

```diff
@@ -0,0 +1,31 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <DataStreams/LimitTransformAction.h>
+#include <Operators/LimitTransformOp.h>
+
+namespace DB
+{
+OperatorStatus LimitTransformOp::transform(Block & block)
+{
+    if (!action->transform(block))
```

Comment:
Is it possible to implement a more abstract transform in TransformOp, which deals invokes transformImpl, and deals with corner cases or end check, something like BlockInputStream.

Reply:
ok, updated.
Has Added `awaitImpl`, `spillImpl`, `transformImpl`, `tryOutputImpl`, `readImpl`, `writeImpl`.